### PR TITLE
fix: initial state error for reverse annealing

### PR DESCRIPTION
**Pull request by jli0108**
*Issue #, if available:* #67

*Description of changes:* 

- Added dependency on `dwave-cloud-client>=0.9.4`.
- Added call to `dwave.cloud.solver.StructuredSolver.reformat_parameters()` before translating from D-Wave format to Braket format.
- Added `self.vartype` in `sample_ising_quantum_task` and `sample_qubo_quantum_task` functions in `BraketDWaveSampler` so that the problem type is available in `_process_solver_kwargs()`. I'm not sure if this is the best way, but it's simple and works.

*Testing done:* I didn't add any new tests, but I ran `tox` successfully.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by shpface** [comment]
Thank you for your contribution!
This will be great to have. There are just a couple changes I would like to see before merging this PR.
* The `reformat_parameters` should be called from both the `sample_qubo_quantum_task` and `sample_qubo`, and same for `ising`
  * To make sure we get this right, I think it may be better to scrap the local variable idea I gave you and call `reformat_parameters` on `kwargs` directly inside each of these 4 methods instead of from `_process_solver_kwargs`.
* Could you add a unit test which passes a dict for `initial_state` to the `BraketDWaveSampler`?

**Comment by jli0108** [comment]
I made the changes you suggested.

File: test/unit_tests/braket/ocean_plugin/conftest.py [hunk 1]
```diff
@@ -319,6 +329,9 @@ def sample_ising_common_testing(
     assert actual.info == info
 
 
+# add test for reverse annealing
+
+
```
[the remark this answers is not in the document]
Comment (jli0108) [reply test/unit_tests/braket/ocean_plugin/conftest.py:334]: Oops, it was a note for myself from yesterday. It should be removed.

**Review by shpface — APPROVED** [review]

**Comment by shpface** [comment]
Great! Thank you!

**Pull request merged**